scylla/src/policies/load_balancing/enforce_target.rs

```diff
@@ -0,0 +1,85 @@
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+use uuid::Uuid;
+
+use crate::cluster::{ClusterState, Node, NodeRef};
+use crate::routing::Shard;
+
+use super::{LoadBalancingPolicy, RoutingInfo};
+
+/// Node identifier used by [`SingleTargetLoadBalancingPolicy`].
+///
+/// If `NodeIdentifier::Node` is used, it will be returned immediately without
+/// the lookup in the cluster metadata.
+///
+/// If `NodeIdentifier::NodeId` or `NodeIdentifier::NodeAddress` is used,
+/// the node will be looked up in the cluster metadata based on its host_id/address.
+#[derive(Debug, Clone)] // <- Cheaply clonable
+#[non_exhaustive]
+pub enum NodeIdentifier {
+    Node(Arc<Node>),
+    NodeId(Uuid),
+    NodeAddress(SocketAddr),
```

Comment:
:thought_balloon: Note that the way it's used implies use of **untranslated** addresses. This must be clearly stated.